packages/search/addon/components/navi-search-bar.js

```diff
@@ -1,3 +1,91 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * Navi Search Bar
+ */
+
 import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { keepLatestTask } from 'ember-concurrency-decorators';
+
+/**
+ * @constant emptyResult – Empty result object
+ */
+const emptyResult = [
+  {
+    title: '',
+    component: 'navi-search-result/no-result'
+  }
+];
+
+export default class NaviSearchBarComponent extends Component {
+  @service('navi-search-provider') searchProviderService;
+
+  /**
+   * @property {String} searchQuery
+   */
+  @tracked searchQuery = '';
+  /**
+   * @property {String} searchResults
+   */
+  @tracked searchResults = [];
+
+  /**
+   * @method search – Perform search based on user query
+   * @param {Object} dd
+   * @param {Object} event
+   */
+  @action
+  search(dd, event) {
+    // Close results window if the user deletes the query or presses escape
+    if (this.searchQuery.length == 0 || event.code === 'Escape') {
```

Comment:
```suggestion
    if (this.searchQuery.length == 0 || event.keyCode === ESCAPE_KEY) {
```

and add `const ESCAPE_KEY = 27;` to top of file